app/src/main/java/com/bogdan/codeforceswatcher/ui/AppRateDialog.kt

```diff
@@ -31,34 +31,42 @@ class AppRateDialog : DialogFragment(), View.OnClickListener {
 
     override fun onClick(view: View) {
         when (view.id) {
-            R.id.btnYes -> {
-                try {
-                    requireContext().startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(GP_MARKET + requireContext().packageName)))
-                } catch (e: Exception) {
-                    Toast.makeText(requireContext(), getString(R.string.play_store_not_found), Toast.LENGTH_SHORT).show()
-                    e.printStackTrace()
-                    dismiss()
-                }
-                prefs.appRated()
-            }
-            R.id.btnNo -> {
-                if (prefs.ratePeriod >= 25) {
-                    prefs.appRated()
-                } else {
-                    prefs.ratePeriod *= 5
-                }
-            }
+            R.id.btnYes -> onClickYes()
+
+            R.id.btnNo -> onClickNo()
         }
         dismiss()
     }
 
+    private fun onClickYes() {
+        try {
+            requireContext().startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(GP_MARKET + requireContext().packageName)))
+        } catch (e: Exception) {
+            Toast.makeText(requireContext(), getString(R.string.play_store_not_found), Toast.LENGTH_SHORT).show()
+            e.printStackTrace()
+            dismiss()
+        }
+        prefs.appRated()
+    }
+
+    private fun onClickNo() {
+        if (prefs.ratePeriod == BIG_PERIOD) {
```

Comment:
Don't you want to dismiss the dialog, when user clicks 'No'?

Reply:
No, I don't)
I dismiss() dialog in onClick() method